patches/MAGECLOUD-2820__implement_isolated_connections_mechanism__2.1.13.patch

```diff
@@ -0,0 +1,434 @@
+-        } catch (\Exception $e) {
+-            error_reporting($phpErrorReporting);
+-            throw $e;
+-        }
+-        return $result;
++        return $this->getDbAdapter()->query($sql);
+     }
+ 
+     /**
+@@ -123,14 +90,44 @@ class MysqlTest extends \PHPUnit_Framework_TestCase
+      *
+      * @return \Magento\Framework\DB\Adapter\Pdo\Mysql
+      */
+-    protected function _getConnection()
++    private function getDbAdapter()
+     {
+-        if (is_null($this->_connection)) {
+-            /** @var $coreResource \Magento\Framework\App\ResourceConnection */
+-            $coreResource = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()
+-                ->get('Magento\Framework\App\ResourceConnection');
+-            $this->_connection = $coreResource->getConnection();
+-        }
+-        return $this->_connection;
++        return $this->resourceConnection->getConnection();
++    }
++
++    public function testGetCreateTable()
++    {
++        $tableName = $this->resourceConnection->getTableName('core_config_data');
++        $this->assertEquals(
++            $this->getDbAdapter()->getCreateTable($tableName),
++            $this->getDbAdapter()->getCreateTable($tableName)
++        );
++    }
++
++    public function testGetForeignKeys()
++    {
++        $tableName = $this->resourceConnection->getTableName('core_config_data');
++        $this->assertEquals(
++            $this->getDbAdapter()->getForeignKeys($tableName),
++            $this->getDbAdapter()->getForeignKeys($tableName)
++        );
++    }
++
++    public function testGetIndexList()
++    {
++        $tableName = $this->resourceConnection->getTableName('core_config_data');
++        $this->assertEquals(
++            $this->getDbAdapter()->getIndexList($tableName),
++            $this->getDbAdapter()->getIndexList($tableName)
++        );
++    }
++
++    public function testDescribeTable()
++    {
++        $tableName = $this->resourceConnection->getTableName('core_config_data');
++        $this->assertEquals(
++            $this->getDbAdapter()->describeTable($tableName),
++            $this->getDbAdapter()->describeTable($tableName)
++        );
+     }
+ }
+diff -Nuar a/vendor/magento/framework/DB/Statement/Pdo/Mysql.php b/vendor/magento/framework/DB/Statement/Pdo/Mysql.php
+--- a/vendor/magento/framework/DB/Statement/Pdo/Mysql.php
++++ b/vendor/magento/framework/DB/Statement/Pdo/Mysql.php
+@@ -3,23 +3,20 @@
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
++namespace Magento\Framework\DB\Statement\Pdo;
+ 
+-// @codingStandardsIgnoreFile
++use Magento\Framework\DB\Statement\Parameter;
+ 
+ /**
+  * Mysql DB Statement
+  *
+  * @author      Magento Core Team <core@magentocommerce.com>
+  */
+-namespace Magento\Framework\DB\Statement\Pdo;
+-
+-use Magento\Framework\DB\Statement\Parameter;
+-
+ class Mysql extends \Zend_Db_Statement_Pdo
+ {
++
```

Comment:
Have you created the patch from fix to core or have you created it after passing the tests?
If so, you should not have this line there.

Reply:
@NadiyaS, I created the patch from fix to core, i did it after passing the tests